src/commands/Ban.tsx

```diff
@@ -0,0 +1,161 @@
+ * This modified file incorperates work from mjolnir
+ * https://github.com/matrix-org/mjolnir
+ * which included the following license notice:
+
+Copyright 2019-2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ *
+ * However, this file is modified and the modifications in this file
+ * are NOT distributed, contributed, committed, or licensed under the Apache License.
+ */
+
+import { MatrixRoomReference } from "./interface-manager/MatrixRoomReference";
+import { UserID } from "matrix-bot-sdk";
+import { MjolnirContext } from "./CommandHandler";
+import { CommandError, CommandResult } from "./interface-manager/Validation";
+import { Mjolnir } from "../Mjolnir";
+import { RULE_ROOM, RULE_SERVER, RULE_USER } from "../models/ListRule";
+import PolicyList from "../models/PolicyList";
+import { defineInterfaceCommand, findTableCommand } from "./interface-manager/InterfaceCommand";
+import { findPresentationType, makePresentationType, ParamaterDescription, paramaters, ParsedKeywords, RestDescription, simpleTypeValidator, union } from "./interface-manager/ParamaterParsing";
+import "./interface-manager/MatrixPresentations";
+import { defineMatrixInterfaceAdaptor } from "./interface-manager/MatrixInterfaceAdaptor";
+import { tickCrossRenderer } from "./interface-manager/MatrixHelpRenderer";
+import { PromptOptions } from "./interface-manager/PromptForAccept";
+import { definePresentationRenderer } from "./interface-manager/DeadDocumentPresentation";
+import { DocumentNode } from "./interface-manager/DeadDocument";
+import { JSXFactory } from "./interface-manager/JSXFactory";
+
+makePresentationType({
+    name: "PolicyList",
+    validator: simpleTypeValidator("PolicyList", (readItem: unknown) => readItem instanceof PolicyList)
+})
+
+definePresentationRenderer(findPresentationType("PolicyList"), function(list: PolicyList): DocumentNode {
+    return <p>
+        {list.listShortcode} <a href={list.roomRef}>{list.roomId}</a>
+    </p>
+})
+
+
+export async function findPolicyListFromRoomReference(mjolnir: Mjolnir, policyListReference: MatrixRoomReference): Promise<CommandResult<PolicyList, CommandError>> {
+    // do we need to catch 404 here so we can tell the user the alias was wrong?
+    // Can we shortcut this by making a specific presentaion type for
+    // resolved room id? (that aliases can translate to)
+    const policyListRoomId = (await policyListReference.resolve(this.client)).toRoomIdOrAlias();
```

Comment:
I would catch it and return a command error.

Reply:
I would agree, but it is rather inconvenient to do without wrapping in try catch with a let or making a `CommandResult` unnecessarily. Quite annoyed. 

```
    const resolvedReference = await policyListReference.resolve(this.client)
        .catch(e => e.statusCode === 404 ? null : Promise.reject(e));
    if (resolvedReference === null) {
        return CommandError.Result(`Failed to resolve the room alias that was provided`);
    }
```

it is possible to do something like this but meh it doesn't seem right